src/h3lib/lib/faceijk.c

```diff
@@ -666,33 +674,34 @@ void _faceIjkPentToVerts(FaceIJK* fijk, int* res, FaceIJK* fijkVerts) {
  *
  * @param h The FaceIJK address of the cell.
  * @param res The H3 resolution of the cell.
- * @param isPentagon Whether or not the cell is a pentagon.
+ * @param start The first topological vertex to return.
+ * @param length The number of topological vertexes to return.
  * @param g The spherical coordinates of the cell boundary.
  */
-void _faceIjkToGeoBoundary(const FaceIJK* h, int res, int isPentagon,
+void _faceIjkToGeoBoundary(const FaceIJK* h, int res, int start, int length,
```

Comment:
There are two main changes to `_faceIjkToGeoBoundary`:
- It now takes `start` and `length` to allow it to efficiently return only a subset of the vertex sequence
- The caller is now responsible for switching between `_faceIjkToGeoBoundary` and `_faceIjkPentToGeoBoundary`, instead of passing in `isPentagon`. This is in part to limit the number of arguments, and in part because requesting all vertexes now requires passing in different lengths for hexes and pentagons.

There was only one other callsite, so this was an easy refactor.